### DISCUSSION TITLE
Exclude files based on text they might contain

### DISCUSSION THREAD

**Discussion by oldergod** [diffplug/spotless#1738] [Q&A]
Hello

In my project, I have a bunch of generated files which are present in many places. They all have a header like `// Generated, do not edit.`. I would like to configure Spotless to ignore those files. How could I do it?

Thank you

**Comment by oldergod** [comment]
I'm giving it a try with `toggleOffOnRegex("(// My Great Header.*)")`

**Reply by oldergod** [reply]
@nedtwigg It looks like this would work but the license is applied before the toggle is used and I get an error `>An intermediate step removed a match of ^(// Code generated[\s\S]*)$`

Is the order on purpose? Would you be open to change it?

**Reply by nedtwigg** [reply]
You want Spotless to skip formatting the entire file, correct?

https://github.com/diffplug/spotless/blob/d245b6086e0aae54b9cdab641e513ab13e06ce6c/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/FormatExtension.java#L852-L856

is a variant of 

https://github.com/diffplug/spotless/blob/d245b6086e0aae54b9cdab641e513ab13e06ce6c/plugin-gradle/src/main/java/com/diffplug/gradle/spotless/FormatExtension.java#L865-L868

which is only for skipping part of a file. To skip a whole file, you'll want to use this somehow

https://github.com/diffplug/spotless/blob/d245b6086e0aae54b9cdab641e513ab13e06ce6c/lib/src/main/java/com/diffplug/spotless/FormatterStep.java#L48-L58

but you'll want to apply it to every single step. I think your request is reasonable and easy to build. The first part is to figure out how to document the feature. Maybe `targetExcludeIfContentMatches(String regex)` or `targetExcludeIfContentContains(String substring)`. Happy to merge a PR for this feature. Even if a PR only has the documentation change (Gradle or Maven or both) maybe somebody else will finish the implementation because it's pretty easy.

**Reply by oldergod** [reply]
I realized that toggling cannot be use because it's only part of a file as you said. I would instead want a file to be ignored indeed if the regex match, so toggle isn't it. I'll try to push a PR.

**Reply by nedtwigg** [reply]
Being implemented at https://github.com/diffplug/spotless/pull/1749
